data_models/data_refinery_models/models.py

```diff
@@ -17,26 +18,101 @@ class Meta:
         abstract = True
 
 
-# This model still has a prototypical status, but I needed something to
-# test with and it's at least in the right ballpark
+class SurveyJob(TimeTrackedModel):
+    source_type = models.CharField(max_length=256)
+
+    # The start time of the query used to replicate
+    replication_started_at = models.DateTimeField(null=True)
+
+    # The end time of the query used to replicate
+    replication_ended_at = models.DateTimeField(null=True)
+
+    # The start time of the job
+    start_time = models.DateTimeField(null=True)
+
+    # The end time of the job
+    end_time = models.DateTimeField(null=True)
+
+    class Meta:
+        db_table = "survey_jobs"
+
+
+class BatchStatuses(Enum):
+    NEW = "NEW"
+    DOWNLOADING = "DOWNLOADING"
+    DOWNLOADED = "DOWNLOADED"
+    PROCESSING = "PROCESSING"
+    PROCESSED = "PROCESSED"
+
+
 class Batch(TimeTrackedModel):
+    survey_job = models.ForeignKey(SurveyJob)
     source_type = models.CharField(max_length=256)
     size_in_bytes = models.IntegerField()
     download_url = models.CharField(max_length=2048)
-    raw_format = models.CharField(max_length=256)
-    processed_format = models.CharField(max_length=256)
-    processor_required = models.IntegerField()
-    accession_code = models.CharField(max_length=256)
+    raw_format = models.CharField(max_length=256, null=True)
+    processed_format = models.CharField(max_length=256, null=True)
+    pipeline_required = models.CharField(max_length=256)
+    accession_code = models.CharField(max_length=32)
+    status = models.CharField(max_length=20)
 
     # This field will denote where in our system the file can be found
-    internal_location = models.CharField(max_length=256)
+    internal_location = models.CharField(max_length=256, null=True)
 
     # This will utilize the organism taxonomy ID from NCBI
     organism = models.IntegerField()
 
-    STATUSES = (
-        ("NEW", "New"),
-        ("DOWNLOADED", "Downloaded"),
-        ("PROCESSED", "Proccessed"),
-    )
-    status = models.CharField(max_length=10, choices=STATUSES)
+    class Meta:
+        db_table = "batches"
+
+
+# This table is used for tacking fields onto a Batch record that would
+# be sparsly populated if it was its own column.
+# I.e. one source may have an extra field or two that are worth tracking
+# but are specific to that source.
+class BatchKeyValue(TimeTrackedModel):
+    batch_id = models.ForeignKey(Batch, on_delete=models.CASCADE)
+    key = models.CharField(max_length=256)
+    value = models.CharField(max_length=256)
+
+    class Meta:
+        db_table = "batch_key_values"
+
+
+class ProcessorJob(TimeTrackedModel):
+    batch_id = models.ForeignKey(Batch, on_delete=models.CASCADE)
+    pipeline_applied = models.CharField(max_length=256)
```

Comment:
I had this as `processor_applied` at some point but I think that a single processor is rarely going to be enough so there will be pipelines of processors which will have identifiers. This field will store the identifier of the processor pipeline which was applied during the job.

As an example a processor pipeline could be described in code like this:
```
processor_pipelines = {
  "EXPRESSION_TO_PCL": ["PROCESSOR_A", "PROCESSOR_B", "PROCESSOR_C"],
  ...
}
```
I could potentially rename this field to `processor_pipeline_applied` if you think that's more clear.